tools/system_libs.py

```diff
@@ -1834,9 +1856,16 @@ def add_library(libname):
   if settings.JS_MATH:
     add_library('libjsmath')
 
-  # to override the normal libc printf, we must come before it
+  # C libraries that override libc must come before it
   if settings.PRINTF_LONG_DOUBLE:
     add_library('libprintf_long_double')
+  # libc_size is a size optimization, and therefore not really important when
+  # MAIN_MODULE=1 (which links in all system libraries, leading to a large
+  # size far bigger than any savings from libc_size). Also, libc_size overrides
+  # parts of libc, which will not link due to --whole-archive in MAIN_MODULE=1
+  # currently.
+  if settings.SHRINK_LEVEL >= 2 and settings.MAIN_MODULE != 1:
+    add_library('libc_size')
```

Comment:
Why not continue to call this `optz` like it was originally named?  I think @juj liked that name and it seems reasonable to me.

Reply:
I don't think there is any particular rule about `-` vs `_`.   I would still suggest leaving the name unchanged... but its up to you.

Reply:
Fair point about it being nice to not change the name. But I checked now and none of the existing names have `-` in them, so that would break the current convention. I think in the long term that convention (`_` in names, `-` to separate variations on top of them) is more important.